src/test/scala/org/broadinstitute/dsde/workbench/sam/azure/AzureRoutesSpec.scala

```diff
@@ -74,19 +72,113 @@ class AzureRoutesSpec extends AnyFlatSpec with Matchers with ScalatestRouteTest
+    val samRoutes = genSamRoutes(createAzurePolicy = false)
+    val newUser = createSecondUser(samRoutes)
+
+    // User has no access
+    val request = GetOrCreatePetManagedIdentityRequest(TenantId("some-tenant"), SubscriptionId("some-sub"), MockCrlService.mockMrgName)
+    Post(s"/api/azure/v1/petManagedIdentity/${newUser.email.value}", request) ~> samRoutes.route ~> check {
       handled shouldBe true
       status shouldEqual StatusCodes.Forbidden
       contentType shouldEqual ContentTypes.`application/json`
     }
   }
 
+  it should "return 403 if the user does not have access to the billing profile" in {
+    val samRoutes = genSamRoutes()
+    // Don't add second user to the spend-profile resource
+    val newUser = createSecondUser(samRoutes, addToSpendProfile = false)
+
+    // User has no access
+    val request = GetOrCreatePetManagedIdentityRequest(TenantId("some-tenant"), SubscriptionId("some-sub"), MockCrlService.mockMrgName)
+    Post(s"/api/azure/v1/petManagedIdentity/${newUser.email.value}", request) ~> samRoutes.route ~> check {
+      handled shouldBe true
+      status shouldEqual StatusCodes.NotFound
+      contentType shouldEqual ContentTypes.`application/json`
+    }
+  }
+
+  private def genSamRoutes(createSpendProfile: Boolean = true, createAzurePolicy: Boolean = true, crlService: Option[CrlService] = None): TestSamRoutes = {
+    val resourceTypes = configResourceTypes.view.filterKeys(k => k == SamResourceTypes.spendProfile || k == CloudExtensions.resourceTypeName)
+    val samRoutes = TestSamRoutes(resourceTypes.toMap, crlService = crlService)
+
+    // Create mock spend-profile resource
+    if (createSpendProfile) {
+      Post(s"/api/resources/v2/${MockCrlService.mockSamSpendProfileResource.resourceTypeName.value}/${MockCrlService.mockSamSpendProfileResource.resourceId.value}") ~> samRoutes.route ~> check {
+        status shouldEqual StatusCodes.NoContent
+      }
+    }
+
+    // Create Azure cloud-extension resource
+    Post(s"/api/resources/v2/${CloudExtensions.resourceTypeName.value}/${AzureExtensions.resourceId}") ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.NoContent
+    }
+
+    // Create azure policy on cloud-extension resource
+    if (createAzurePolicy) {
+      val cloudExtensionMembers = AccessPolicyMembership(Set(samRoutes.user.email), Set(AzureExtensions.getPetManagedIdentityAction), Set.empty, None)
+      Put(s"/api/resources/v2/${CloudExtensions.resourceTypeName.value}/${AzureExtensions.resourceId.value}/policies/azure", cloudExtensionMembers) ~> samRoutes.route ~> check {
+        status shouldEqual StatusCodes.Created
+      }
+    }
+
+    samRoutes
+  }
+
+  private def createSecondUser(samRoutes: TestSamRoutes, addToSpendProfile: Boolean = true): SamUser = {
+    val newUser = Generator.genWorkbenchUserGoogle.sample.get
+    IO.fromFuture(IO(samRoutes.userService.createUser(newUser, samRequestContext))).unsafeRunSync()
```

Comment:
probably can just do `samRoutes.userService.createUser(newUser, samRequestContext)` here? looks like you want the side effect to happen here

Reply:
`createUser` returns a `Future`, I want to wait for it to complete… I guess I could use `Await(futre)` instead of converting to IO..